security/src/main/java/nl/nn/adapterframework/lifecycle/servlets/ServletAuthenticatorBase.java

```diff
@@ -103,8 +115,13 @@ private void addEndpoints(List<String> urlMappings) {
 				log.info("registering public endpoint with url [{}]", publicUrl);
 				publicEndpoints.add(publicUrl);
 			} else {
```

Comment:
```suggestion
			} else if(config.getSecurityRoles().isEmpty()) {
	            log.info("registering public endpoint with url pattern [{}]", url);
	            publicEndpoints.add(url);
            } else {
	            log.info("registering private endpoint with url pattern [{}]", url);
	            privateEndpoints.add(url);
            }
```

Reply:
is de huidge opzet met in het else block weer een if/else statement echt nodig?

Reply:
De eerste if past de url aan, deze zet hem direct.

Reply:
Dat snap, ik geef ook niet de suggestie om de if's te combineren...

![image](https://github.com/frankframework/frankframework/assets/37303445/17427d27-cb25-476a-bbfc-1fccd031f2a2)


Naar mijn idee is het if/else statement in het else block overbodig and kan het als volgt geschreven worden:

```
if(url.charAt(0) == '!') {
				String publicUrl = url.substring(1);
				log.info("registering public endpoint with url [{}]", publicUrl);
				publicEndpoints.add(publicUrl);
			} else if(config.getSecurityRoles().isEmpty()) {
			    log.info("registering public endpoint with url pattern [{}]", url);
			    publicEndpoints.add(url);
		    } else {
			    log.info("registering private endpoint with url pattern [{}]", url);
			    privateEndpoints.add(url);
		    }	
```

effectief doet dit hetzelfde...

Reply:
Ik denk dat
```java
boolean isExcludedUrl = url.charAt(0) == '!';
if(isExcludedUrl || config.getSecurityRoles().isEmpty()) {
	String publicUrl = isExcludedUrl ? url.substring(1) : url;
	log.info("registering public endpoint with url [{}]", publicUrl);
	publicEndpoints.add(publicUrl);
} else {
	log.info("registering private endpoint with url pattern [{}]", url);
	privateEndpoints.add(url);
}
```
dan misschien nog net iets netter is. Ik zal dit in een volgend pr meenemen.